src/main/lombok/eu/coinform/gateway/events/PasswordResetListener.java

```diff
@@ -0,0 +1,28 @@
+package eu.coinform.gateway.events;
+
+import eu.coinform.gateway.db.entity.User;
+import eu.coinform.gateway.db.UserDbManager;
+import eu.coinform.gateway.service.EmailService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class PasswordResetListener extends GatewayEventListener<OnPasswordResetEvent>{
+
+    private final UserDbManager userDbManager;
+
+    PasswordResetListener(UserDbManager userDbManager,
+                          EmailService emailService) {
+        super(emailService);
+        this.userDbManager = userDbManager;
+    }
+
+    @Override
+    protected void handleEvent(OnPasswordResetEvent event) {
+        User user = event.getUser();
+        String token = userDbManager.passwordReset(user);
+        String verifyUrl = url + "/passwordreset?token="+token;
```

Comment:
I thought about that and using the email as part of the url exposes the users email to any machine that parses the url for fetching the webpage for the user (since the url itself is not encrypted). Doing it this way does not. The token itself is unique enough to identify the user.

Reply:
Queries are encrypted in https connections

Reply:
The whole url is encrypted in the under the tls connection its only the preceding dns query that's unencrypted

Reply:
Its only the initial handshake to the api.gateway.eu server that is unencrypted